rotkehlchen/chain/evm/manager.py

```diff
@@ -0,0 +1,1008 @@
+    deserialize_evm_transaction,
+    deserialize_int_from_hex,
+)
+from rotkehlchen.serialization.serialize import process_result
+from rotkehlchen.types import (
+    ChecksumEvmAddress,
+    EvmTransaction,
+    EVMTxHash,
+    SupportedBlockchain,
+    Timestamp,
+)
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.misc import from_wei, get_chunks, hex_or_bytes_to_str
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _is_synchronized(current_block: int, latest_block: int) -> Tuple[bool, str]:
+    """ Validate that the evm node is synchronized
+            within 20 blocks of latest block
+
+        Returns a tuple (results, message)
+            - result: Boolean for confirmation of synchronized
+            - message: A message containing information on what the status is.
+    """
+    message = ''
+    if current_block < (latest_block - 20):
+        message = (
+            f'Found evm node but it is out of sync. {current_block} / '
+            f'{latest_block}. Will use etherscan.'
+        )
+        log.warning(message)
+        return False, message
+
+    return True, message
+
+
+WEB3_LOGQUERY_BLOCK_RANGE = 250000
+
+
+class EvmManager:
+    """EvmManager defines a basic implementation for most of EVM chains."""
+    def __init__(
+            self,
+            etherscan: Etherscan,
+            msg_aggregator: MessagesAggregator,
+            greenlet_manager: GreenletManager,
+            connect_at_start: Sequence[WeightedNode],
+            database: 'DBHandler',
+            etherscan_node: WeightedNode,
+            etherscan_node_name: str,
+            blockchain: SupportedBlockchain,
+            chain_name: str,
+            chain_id: ChainID,
+            contract_scan: EvmContract,
+            contract_multicall: EvmContract,
+            contract_multicall_2: EvmContract,
+            rpc_timeout: int = DEFAULT_EVM_RPC_TIMEOUT,
+    ) -> None:
+        self.greenlet_manager = greenlet_manager
+        self.web3_mapping: Dict[NodeName, Web3] = {}
+        self.etherscan = etherscan
+        self.msg_aggregator = msg_aggregator
+        self.database = database
+        self.etherscan_node = etherscan_node
+        self.etherscan_node_name = etherscan_node_name
+        self.blockchain = blockchain
+        self.chain_name = chain_name
+        self.chain_id = chain_id
+        self.contract_scan = contract_scan
+        self.contract_multicall = contract_multicall
+        self.contract_multicall_2 = contract_multicall_2
+        self.rpc_timeout = rpc_timeout
+        self.archive_connection = False
+        self.queried_archive_connection = False
+        self.connect_to_multiple_nodes(connect_at_start)
+
+        # TODO: Remove soon
+        self.blocks_subgraph = Graph(
+            'https://api.thegraph.com/subgraphs/name/blocklytics/ethereum-blocks',
+        )
```

Comment:
This url should also be configurable. We want to remove dependencies from thegraph but this should be configurable in the arguments

Reply:
Done! :1+: